### DISCUSSION TITLE
Should `ignore` be respected in `version` and `publish`?

### DISCUSSION THREAD

**Discussion by denolfe** [ghiscoding/lerna-lite#823] [Q&A] [answered]
I have a monorepo setup with pnpm, and I am trying to ignore a specific package for all commands.

I've setup a recreation like this:

lerna.json
```
{
  "$schema": "node_modules/@lerna-lite/cli/schemas/lerna-schema.json",
  "version": "1.0.0",
  "verbose": true,
  "ignore": "ignore-me",
  "packages": [
    "packages/*"
  ]
}
```

Directory structure
```
.
├── lerna.json
├── package.json
├── packages
│   ├── ignore-me
│   │   ├── index.js
│   │   └── package.json
│   └── package-1
│       ├── index.js
│       └── package.json
├── pnpm-lock.yaml
└── pnpm-workspace.yaml
```

Running `lerna list` properly respects `ignore`

```
❯ pnpm lerna list
lerna-lite notice cli v3.3.0
lerna-lite verb rootPath /Users/elliot/dev/sandbox/lerna-monorepo-test
lerna-lite verb project packages packages/*
lerna-lite notice filter excluding "ignore-me"
lerna-lite info filter ["!ignore-me"]
package-1
lerna-lite success found 1 package
```

However, running `lerna version` disregards `ignore`

```
❯ pnpm lerna version --dry-run --bump premajor
lerna-lite notice cli v3.3.0
lerna-lite verb rootPath /Users/elliot/dev/sandbox/lerna-monorepo-test
lerna-lite info [dry-run] > git rev-parse
lerna-lite verb project packages packages/*
lerna-lite info current project version 1.0.0
lerna-lite info [dry-run] > git rev-list --count --all --max-count=1
lerna-lite verb isAnythingCommitted 1
lerna-lite verb currentBranch main
lerna-lite info [dry-run] > git show-ref --verify refs/remotes/origin/main
lerna-lite info [dry-run] > git remote update
lerna-lite info [dry-run] > git rev-list --left-right --count origin/main...main
lerna-lite verb hasTags true
lerna-lite info Looking for changed packages since v0.0.1
lerna-lite verb filtered diff packages/ignore-me/index.js packages/ignore-me/package.json
lerna-lite verb filtered diff packages/package-1/index.js packages/package-1/package.json
lerna-lite verb updated ignore-me
lerna-lite verb updated package-1
lerna-lite info [dry-run] > git describe --always --long --dirty --first-parent

Changes (2 packages):
 - ignore-me: 1.0.0 => 2.0.0-alpha.0
 - package-1: 1.0.0 => 2.0.0-alpha.0
```

Is this expected behavior?

I've tried also adding `ignore` inside the `list` command settings, but this did not make a difference either.

**Comment by ghiscoding** [comment]
I have never used that option but looking at the [@lerna-lite/filter-packages](https://github.com/lerna-lite/lerna-lite/tree/main/packages/filter-packages) it seems to only cover `exec`, `list`, `run` and `watch` (probably `diff` too) and does not seem to cover `publish` and `version`. That code came from Lerna so I suspect that they've never added this filtering to version/publish

![image](https://github.com/lerna-lite/lerna-lite/assets/643976/11fbd271-82e3-45fb-a720-50e7f4a2d9a9)

**Reply by ghiscoding** [reply, marked as answer]
looking at Lerna's issue, that seems to be the case as per this [issue](https://github.com/lerna/lerna/issues/3874) which is pretty much what you're asking. Searching on the internet and I also stumble across similar answer that I wrote myself (I forgot about that one lol), see https://stackoverflow.com/a/74184091/1212166

Perhaps you can try `lerna version --ignore-changes 'packages/ignore-me/**'` (see Lerna [comment](https://github.com/lerna/lerna/issues/1586#issuecomment-434903591))

**Reply by denolfe** [reply]
Thanks @ghiscoding for taking a look. I searched the upstream issues before making this post but didn't turn that one up. 

I'll weigh in on that issue 👍
